routing/cross_mwm_connector.cpp

```diff
@@ -54,7 +54,8 @@ bool CrossMwmConnector::IsTransition(Segment const & segment, bool isOutgoing) c
   if (transition.m_oneWay && !segment.IsForward())
     return false;
 
-  return (segment.IsForward() == transition.m_forwardIsEnter) == isOutgoing;
+  bool const isEnter = (segment.IsForward() == transition.m_forwardIsEnter);
```

Comment:
Done

```
  // Note. If |isOutgoing| == true |segment| should be an exit transition segment
  // (|isEnter| == false) to a transition segment.
  // Otherwise |segment| should be an enter transition segment (|isEnter| == true)
  // to be a trasition segment. If not, |segment| is not a transition segment.
  // Please see documentation on CrossMwmGraph::IsTransition() method for details.
  bool const isEnter = (segment.IsForward() == transition.m_forwardIsEnter);
  return isEnter != isOutgoing;
```